### PR TITLE
 Fix item id misspellings

### LINKED ISSUE #46289: Item ID typos and spelling errors for batch migration (add yours)
# Describe the bug

Some item IDs have errors in them. Items have a way to migrate them to new IDs via `data/json/items/migration.json`.

I suggest we collect all the IDs and migrate them in batch.

ID | suggested replacement
-- | --
balclava | balaclava
lucern_hammer | lucerne_hammer
bowhat | bowlhat, hat_bowl, hat_bowler
porkpie | hat_pork_pie
roller_blades | rollerblades

Vehicle parts, if a migration is implemented:

VP ID | suggested replacement
-- | --
wood box | wood_box
foldable wood box | foldable_wood_box

Bionics

ID | suggested replacement
-- | --
mend_fault_bonic_salvaged_reset | mend_fault_bionic_salvaged_reset

# Additional context

#45878 #44460 #37967 #38073

### PR DESCRIPTION
**Pull request by hexagonrecursion**
#### Summary

SUMMARY: Infrastructure "Rename misspelled item ids"

#### Purpose of change

old id | new id
-|-
balclava | balaclava
lucern_hammer | lucerne_hammer
wood box | wood_box
folding wood box | folding_wood_box
bowhat | hat_bowler
porkpie | hat_pork_pie
roller_blades | rollerblades
ax | axe

Fixes #46289

#### Describe the solution

1. Try to automatically find and replace all occurrences of misspelled item ids (see commits for details).
2. Migrate old saves via data/json/items/migration.json

#### Describe alternatives you've considered

leave as is?

#### Testing

1. The game loads without errors.
2. rollerblades have the new id in the wish menu
3. roller_blades are successfully migrated to rollerblades when loading an old save
4. An old save with a car with a wood box does not explode the game

#### Additional context

none

### CODE REVIEW

File: data/json/items/migration.json [hunk 1]
```diff
@@ -1313,5 +1313,55 @@
     "id": "rifle_3006",
     "type": "MIGRATION",
     "replace": "remington_700"
+  },
+  {
+    "id": "balclava",
+    "type": "MIGRATION",
+    "replace": "balaclava"
```
Comment (int-ua) [inline data/json/items/migration.json:1320]: Not sure if it was discussed before, should there also be comments (`"\\"` JSON key) about version the migration was added in? In this case, `0.F`.

Comment (hexagonrecursion) [reply data/json/items/migration.json:1320]: 1. I don't see any of the other migrations having this comments
2. Isn't git blame sufficient?

Comment (int-ua) [reply data/json/items/migration.json:1320]: 1. It's just a suggestion that came to my mind minutes ago :)
2. Definitely sufficient but makes determining what migrations should be dropped for which version harder and can be made even harder if JSON style ever changes and it's re-linted or something similar.

**Comment by int-ua** [comment]
(In case you haven't noticed yet) Tests also have some item IDs hardcoded https://github.com/CleverRaven/Cataclysm-DDA/search?l=C%2B%2B&q=balclava

**Comment by hexagonrecursion** [comment]
I had no idea

**Comment by int-ua** [comment]
May I ask listing changed IDs explicitly and adding the Summary as this change affects tileset makers and needs to be noticeable?

**Comment by mlangsdorf** [comment]
I'm really opposed to this idea.  This is churn for churn's sake, and probably breaks the effective DPS tests.  There's no reason to change the item IDs for the various axes or the Lucerne Hammer - it doesn't matter if they're not spelled consistently or if `lucern_hammer` isn't spelled "correctly".  It's an arbitrary token.

**Comment by int-ua** [comment]
I believe tests are fixed now. And while `ax`/`axe` and spaces in `id` changes are closer to preference, the `balclava` and `lucern_hammer` are definitely typos. `rollerblades` being somewhere in the middle.

These IDs are almost equivalent to variable names in code in terms of readability. And readability matters. Obfuscation is a thing precisely because it's important to have readable and easily understandable identifiers.

**Comment by int-ua** [comment]
`0.7623762376 < 0.6f` The one failure doesn't look related to these changes, the same test passed successfully in other build matrix instances.

File: lang/po/ar.po [hunk 2]
```diff
@@ -255800,11 +255800,11 @@ msgid "travois"
 msgstr ""
 
 #: lang/json/vehicle_part_from_json.py
-msgid "wood box"
+msgid "wood_box"
```
Comment (int-ua) [inline lang/po/ar.po:255803]: I think the translation files do not contain ids, it must be the name

Comment (int-ua) [reply lang/po/ar.po:255803]: **May** be reverted easily by `git checkout lang/po/`

File: data/json/items/migration.json (hunk 1 above)
Comment (int-ua) [reply data/json/items/migration.json:1320]: Anyway, feel free to resolve this without changes, I don't mind

File: lang/po/ar.po (hunk 2 above)
Comment (Qrox) [reply lang/po/ar.po:255803]: That's right, and translation files should not be manually edited anyway.

Comment (hexagonrecursion) [reply lang/po/ar.po]: Done

**Comment by int-ua** [comment]
@mlangsdorf I've provided counterarguments, please review them.

**Comment by actual-nh** [comment]
Another one: bio_taste_blocker -> bio_taste_modifier; see #47635.

**Comment by hexagonrecursion** [comment]
Closed: I lost interest in continuing work on this PR.

**Pull request closed without merging**